### PR TITLE
ADM-997 [frontend]: add new label for new functions

### PR DESCRIPTION
**Pull request by zhou-yinyuan**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

image

image

image

image

## Note

_Null_

### CODE REVIEW

File: frontend/src/containers/ReportStep/DoraMetricsChart/index.tsx [hunk 1]
```diff
@@ -325,12 +326,21 @@ export const DoraMetricsChart = ({
...
-        value={selectedPipeline}
-        onUpDatePipeline={(value) => onUpdatePipeline(value)}
-        title={'Pipeline/Step'}
-      />
+      <NewFunctionsLabel
+        style={{
+          margin: '1rem 0 0',
+          height: '2rem',
+          lineHeight: '2rem',
+        }}
```
Comment (guzhongren) [inline frontend/src/containers/ReportStep/DoraMetricsChart/index.tsx:334]: move this part into `styles.tsx`

File: frontend/src/constants/commons.ts [hunk 2]
```diff
@@ -107,3 +107,7 @@ export const ANIMATION_SECONDS = 0.5;
 export const EVERY_FRAME_MILLISECOND = 17;
 
 export const MILLISECONDS_PER_SECOND = 1000;
+
+export const NEW_FUNCTIONS_VERSIONS = {
```
Comment (guzhongren) [inline frontend/src/constants/commons.ts:111]: remove it and use `1.3.0` for all changes directly. 
we can't define every version in the {}

File: frontend/__tests__/containers/ReportStep/DoraMetrics.test.tsx [hunk 3]
```diff
@@ -99,4 +101,21 @@ describe('Report Card', () => {
...
+  it('should show four new labels in the dora metrics when version is less than 1.3.0', () => {
+    setup();
+
+    const newLabels = screen.queryAllByLabelText('new label');
+
+    expect(newLabels.length).toEqual(4);
+    newLabels.forEach((it) => expect(it).toBeInTheDocument());
+  });
+
+  it('should not show any new label in the dora metrics when version is more than 1.3.0', () => {
```
Comment (guzhongren) [inline frontend/__tests__/containers/ReportStep/DoraMetrics.test.tsx:114]: how to compare `1.0.0-alpha、1.0.0-alpha.1、1.0.0-0.3.7、1.0.0-x.7.z.92。`

ref to https://semver.org/lang/zh-CN/

File: frontend/src/components/Common/NewFunctionsLabel/index.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  createVersion: string;
+}
+
+export default function ({ children, style, createVersion }: NewFunctionsLabelProps) {
+  const version = useAppSelector(getVersion);
+
+  return (
+    <NewFunctionsContent>
+      {children}
+      {version <= createVersion && (
```
Comment (guzhongren) [inline frontend/src/components/Common/NewFunctionsLabel/index.tsx:18]: `version <= createVersion` is correct? 
this comparison is based on the `string`， it is not workable

File: frontend/src/components/Common/NewFunctionsLabel/index.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { NewFunctionsContent, NewLabel } from '@src/components/Common/NewFunctionsLabel/style';
+import { getVersion } from '@src/context/meta/metaSlice';
+import { useAppSelector } from '@src/hooks';
+import React, { ReactNode } from 'react';
+
+interface NewFunctionsLabelProps {
+  children: ReactNode;
+  style?: object;
+  createVersion: string;
```
Comment (guzhongren) [inline frontend/src/components/Common/NewFunctionsLabel/index.tsx:9]: rename to `initVersion`

File: frontend/__tests__/components/Common/NewFunctionsLabel.test.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import { render, screen } from '@testing-library/react';
+import { Provider } from 'react-redux';
+
+const store = setupStore();
+
+describe('NewFunctionsLabel', () => {
+  const setup = (version: string, createVersion: string) => {
+    store.dispatch(saveVersion(version));
+    return render(
+      <Provider store={store}>
```
Comment (guzhongren) [inline frontend/__tests__/components/Common/NewFunctionsLabel.test.tsx:13]: how to compare 1.0.0-alpha、1.0.0-alpha.1、1.0.0-0.3.7、1.0.0-x.7.z.92。

ref to https://semver.org/lang/zh-CN/

**Review by guzhongren — COMMENTED** [review]
some comments

File: frontend/src/constants/commons.ts (hunk 2 above)
Comment (zhou-yinyuan) [reply frontend/src/constants/commons.ts:111]: done

File: frontend/src/components/Common/NewFunctionsLabel/index.tsx (hunk 5 above)
Comment (zhou-yinyuan) [reply frontend/src/components/Common/NewFunctionsLabel/index.tsx:9]: done

File: frontend/__tests__/components/Common/NewFunctionsLabel.test.tsx (hunk 6 above)
Comment (zhou-yinyuan) [reply frontend/__tests__/components/Common/NewFunctionsLabel.test.tsx:13]: done

File: frontend/src/components/Common/NewFunctionsLabel/index.tsx (hunk 4 above)
Comment (zhou-yinyuan) [reply frontend/src/components/Common/NewFunctionsLabel/index.tsx:18]: done

File: frontend/__tests__/containers/ReportStep/DoraMetrics.test.tsx (hunk 3 above)
Comment (zhou-yinyuan) [reply frontend/__tests__/containers/ReportStep/DoraMetrics.test.tsx:114]: done

File: frontend/src/containers/ReportStep/DoraMetricsChart/index.tsx (hunk 1 above)
Comment (zhou-yinyuan) [reply frontend/src/containers/ReportStep/DoraMetricsChart/index.tsx:334]: done

**Review by guzhongren — APPROVED** [review]

**Pull request merged**